### ISSUE TITLE
Multi_GPUs cause a lower meanIOU

### ISSUE THREAD

**Issue by TeeyoHuang** [pytorch/pytorch#6095]
PyTorch GitHub Issues Guidelines
--------------------------------
The net I used is deeplab-resnet,and the code is here:
https://github.com/speedinghzl/Pytorch-Deeplab,

It does have a good result when I run the code on 1 GPU， I got the meanIOU as reported-- about 72%  on the validation dataset(VOC12 VAL）

But when I used 5 GPUs to have a test, I got a lower meanIOU --- about 65%.
(I used 5 gpus because the batch_size is 10, so each gpu can process  2 images)

I just add 2 lines:
    Mynet = torch.nn.DataParallel(model.cuda(gpu),device_ids=[0,1,2,3,4])

and the trainning stage: 
    out=Mynet(images.cuda(gpu))

I have tried many times but it still gets the 65% meanIOU, I think maybe I lost some necessary  steps which I don't know, can anybody give a solution? :-)

**Comment by fcheng-cv** [comment]
May be your BN layer is not synchronized version? Small batch size is sensitive to the mean and var values. For semantic segmentation, especially reproducing PSPNet, synchronized batch norm is necessary.
FYI. http://hangzh.com/PyTorch-Encoding/notes/syncbn.html

**Comment by SsnL** [comment]
As @fcheng-cv mentioned, it is likely a BN issue. Tracking stats on devices are not synchronized. If you believe the issue is in elsewhere, please give us more details of the GPUs and how you run the script.

**Comment by soumith** [comment]
closed via comment https://github.com/pytorch/pytorch/issues/6095#issuecomment-378023755
